torch/lib/c10d/reducer.cpp

```diff
@@ -556,10 +556,16 @@ void Reducer::autograd_hook(VariableIndex index) {
         "then got used in the second iteration. this is not ",
         "compatible with static_graph set to True.");
     if (--numGradHooksTriggeredMapPerIteration_[index] == 0) {
+      if (should_rebuild_buckets()) {
+        push_rebuilt_params(index);
+      }
       // Finally mark variable for which this function was originally called.
       mark_variable_ready(index);
     }
   } else {
+    if (should_rebuild_buckets()) {
```

Comment:
Can we just have it before `mark_variable_ready` to reduce the code duplication?

Reply:
the codes need to be put in different places for static graph and non-static graph case. e.g., when grad hooks are triggered more than once in static graph, we want to push this parameter to 'rebuilt_bucket_indices' only once